File: Data Structures & Algorithms/longest-consecutive-sequence/submission-5.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> freq(nums.begin(), nums.end());
        int ans = 0;
        vector<int> starts;
        for (int i = 0; i < nums.size(); i++) {
            if (!freq.count(nums[i] - 1)) {
                int dist = 1;
                while (freq.find(nums[i] + dist) != freq.end()) {
                    dist ++;
                }
                ans = max(dist, ans);
            }
        }
        return ans;
    }
};

File: Data Structures & Algorithms/climbing-stairs/submission-5.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n == 1:
            return 1
        if n == 2:
            return 2
        ans = [0] * n
        counter = n - 1
        ans[counter] = 1
        counter = n - 2
        ans[counter] = 2
        counter = n - 3
        while counter >= 0:
            ans[counter] = ans[counter + 1] + ans[counter + 2]
            counter -= 1
        return ans[0]